cni-plugin/integration/testutil/test_util.go

```diff
@@ -39,6 +40,70 @@ func (r *TestRunner) walkConfDir() (map[string]struct{}, error) {
 	return fileNames, nil
 }
 
+// Checks that the embedded linkerd config is of the expected form
+// and contains the right values:
+//
+//	"linkerd": {
+//	   "incoming-proxy-port": 4143,
+//	   "outgoing-proxy-port": 4140,
+//	   "proxy-uid": 2102,
+//	   "ports-to-redirect": [],
+//	   "inbound-ports-to-ignore": ["4191","4190"],
+//	   "simulate": false,
+//	   "use-wait-flag": false
+//	 }
+func checkLinkerdCniConf(wrapperConf map[string]any) error {
+	var conf = wrapperConf["linkerd"].(map[string]any)
+
+	var incomingProxyPort = conf["incoming-proxy-port"].(float64)
```

Comment:
Genuine question: is there a difference to using a float64 for the port as opposed to using an int?

Reply:
I think this is because JSON only has floating point numbers. When I try to use `int` I get the following error

`panic: interface conversion: interface {} is float64, not int [recovered]`

Reply:
eesh :(